GWXtreme/parametrized_eos_sampler.py

```diff
@@ -179,7 +179,61 @@ def save_data(self):
         f.create_dataset('logp',data=np.array(self.logp))
         f.close()
         
+    def parse_samples(self,burn_in_frac = 0.5, thinning = None):
+        '''
+        This methods loads a .h5 file with the MCMC samples 
+        of EoS hyper-parameters and parses them for use by the 
+        plotting functions.
+        see https://emcee.readthedocs.io/en/stable/tutorials/autocorr/
+        for some documentation on choosing thinning and burn-in
+        
+        burn_in_frac  :: fraction of samples to discard from each chain
+                         for MCMC burn in. Default corresponds to discarding 
+                         half the samples in each chain.
+        
+        thinning      :: Number of samples to skip in each chain post
+                         burn in. "None" implements the default value 
+                         which is either (length of chain)/50 or half of
+                         the maximum integrated autocorrelation time. The 
+                         former is used in case autocorrelation analysis 
+                         yields throws non-convergence error. For no thinning 
+                         set thinning=1
+        
+        '''
+        
+        Ns=self.samples.shape
+        burn_in=int(Ns[0]*burn_in_frac)
+        samples=[]
+        
+        if thinning is None:
+            thinning=int(Ns[0]/50.)
 
+            try:
+                thinning=int(max(mc.autocorr.integrated_time(self.samples))/2.)
+            except mc.autocorr.AutocorrError as e:
+                print(e)
+        
+        for i in range(burn_in,Ns[0],thinning):
+            for j in range(Ns[1]):
+                samples.append(self.samples[i,j,:])
```

Comment:
Is the goal here to keep a thinned array of the samples starting from the burned-in point? In that case, why not simply first select the samples that are after burn-in and then use `np.random.choice` to select the `thinning` number of samples? It will be a much more `Numpy` way of doing things.

Reply:
I think the goal of thinning is to take every nth sample from the chain to make sure that the remaining samples are iid draws. If the pre-thinning chains were fully iid draws then random.choice would amount to the same thing but in such a scenario, thinning would not be needed at all! The way this is done is the same as how emcee thins itself.

Reply:
I may be overly simplifying this in my mind. But why can't this be done by simple Numpy slicing? For example, let's say there are 120 samples with each sample having 4 quantities. Let's say the first twenty points are burn-in, so we discrd them. Then we want to keep the remaining skipping every `n` samples (which is our thinning parameter).

Example:
```
x = np.random.randn(120, 4)
x_burned = x[20:]
x_thinned = x_burned[1::5]
```
The final quantity should be the thinned samples with the desired frequency of sampling. Isn't it just that?

Reply:
Well there is another dimension, the number of walkers. so each walker would be a 120,4 array and there would be 100 such arrays. It might still be possible to do it via slicing, but I was thinking since this is a very fast post-processing step with the benefit from vectorization negligible, is it really worth re-reviewing ? (all the other changes are stylistic).